### DISCUSSION TITLE
Adding a transformation to a page that has content linked from a table of contents removes the content from the page

### DISCUSSION THREAD

**Discussion by glynam1** [py-pdf/pypdf#2300] [Q&A] [answered]
When scaling down content to fit it onto an a4 page, we noticed an issue where if a user has used the table of contents functionality to generate links to pages, each page that contains one of the "destination" links, will end up blank after the transformation.  

## Environment
reproducible on python 3.9 and pypdf 3.17.1

## Code + PDF

```import shutil
import tempfile

from pypdf import PdfReader, PdfWriter, Transformation


def scale_pdf():
    _, temp_output_path = tempfile.mkstemp(suffix=".pdf")
    reader = PdfReader("example.pdf")
    writer = PdfWriter()
    for page in reader.pages:
        t = Transformation()
        t = t.scale(1, 1)
        page.add_transformation(t)
        writer.add_page(page)
    writer.write(temp_output_path)
    shutil.move(temp_output_path, "result.pdf")

if __name__ == '__main__':
    scale_pdf()

```
[example.pdf](https://github.com/py-pdf/pypdf/files/13347986/example.pdf)

[result.pdf](https://github.com/py-pdf/pypdf/files/13347990/result.pdf)

**Comment by glynam1** [comment]
This is a regression from 3.11.0, testing on 3.10.0 fixes the issue

**Comment by pubpub-zz** [comment, marked as answer]
using add_page you are loosing the link between pages that do exist in the source.
this is the way I propose to do it:
```
import pypdf
w=pypdf.PdfWriter(clone_from="example.pdf")
for p in w.pages:
    t=pypdf.Transformation().scale(1,1)
    p.add_transformation(t)
w.write("result.pdf")
```

**Comment by glynam1** [comment]
> using add_page you are loosing the link between pages that do exist in the source. this is the way I propose to do it:
> 
> ```
> import pypdf
> w=pypdf.PdfWriter(clone_from="example.pdf")
> for p in w.pages:
>     t=pypdf.Transformation().scale(1,1)
>     p.add_transformation(t)
> w.write("result.pdf")
> ```


This works great, thanks for that!
